DataFormats/Detectors/MUON/MID/src/Track.cxx

```diff
@@ -116,9 +112,9 @@ bool Track::propagateToZ(float zPosition)
 bool Track::isCompatible(const Track& track, float chi2Cut) const
 {
   /// Check if tracks are compatible within uncertainties
-  if (track.getPosition().z() != getPosition().z()) {
+  if (track.mPosition[2] != mPosition[2]) {
```

Comment:
as above for equality. Don't you want to use some tolerance here ?

Reply:
Same comment as above, since I extrapolate to Z which is typically the position of the cluster.